examples/tutorials/mvdr_tutorial.py

```diff
@@ -38,258 +52,315 @@
-# Multi-channel mask results
-# ~~~~~~~~~~~~~~~~~~~~~~~~~~
+# 3.4.2. Visualize IRM of noise
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 #
 
-for solution in results_multi:
-    print(solution + ": ", si_sdr(results_multi[solution][None, ...], reverb_clean[0:1]))
+plot_mask(irm_noise, "IRM of the Noise")
 
-######################################################################
-# Original audio
-# --------------
-#
 
 ######################################################################
-# Mixture speech
-# ~~~~~~~~~~~~~~
+# 4. Beamforming using SoudenMVDR
+# -------------------------------
 #
 
-ipd.Audio(mix[0], rate=16000)
 
 ######################################################################
-# Noise
-# ~~~~~
+# 4.1. Compute PSD matrices
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 #
+# :py:func:`torchaudio.transforms.PSD` computes the time-invariant PSD matrix given
+# the multi-channel complex-valued spectrum and the Time-Frequency mask.
+#
+# The shape of the PSD matrix is `(..., freq, channel, channel)`.
+
+psd_transform = torchaudio.transforms.PSD()
+
+psd_speech = psd_transform(spectrum_mix, irm_speech)
+psd_noise = psd_transform(spectrum_mix, irm_noise)
 
-ipd.Audio(noise[0], rate=16000)
 
 ######################################################################
-# Clean speech
-# ~~~~~~~~~~~~
+# 4.2. Apply Beamforming
+# ~~~~~~~~~~~~~~~~~~~~~~
+#
+# :py:func:`torchaudio.transforms.SoudenMVDR` takes the multi-channel
+# complexed-valued spectrum of the mixture speech, PSD matrices of
+# target speech and noise, and the reference channel as the inputs.
 #
+# The output is a single-channel complex-valued spectrum of the enhanced speech.
+# Then we can obtain the enhanced wavefrom by passing it to the
+# :py:func:`torchaudio.transforms.InverseSpectrogram` module.
+
+mvdr_transform = torchaudio.transforms.SoudenMVDR()
+spectrum_souden = mvdr_transform(spectrum_mix, psd_speech, psd_noise, reference_channel=REFERENCE_CHANNEL)
+waveform_souden = istft(spectrum_souden, length=waveform_mix.shape[-1])
 
-ipd.Audio(clean[0], rate=16000)
 
 ######################################################################
-# Enhanced audio
-# --------------
+# 4.3. Result for SoudenMVDR
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
 #
 
+plot_specgram(spectrum_souden, "Enhanced Spectrogram by SoudenMVDR (dB)")
+waveform_souden = waveform_souden.reshape(1, -1)
+print(f"Si-SNR score: {si_snr(waveform_souden, waveform_clean[0:1])}")
+Audio(waveform_souden, rate=SAMPLE_RATE)
+
+
 ######################################################################
-# Multi-channel mask, ref_channel solution
-# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# 5. Beamforming using RTFMVDR
+# ----------------------------
 #
 
-ipd.Audio(results_multi["ref_channel"], rate=16000)
 
 ######################################################################
-# Multi-channel mask, stv_evd solution
-# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# 5.1. Compute PSD matrices
```

Comment:
to maintain parallelism with the steps you outlined in the beginning, where you identify PSD matrix generation as a distinct step, i'd break this (and the corresponding section for `SoudenMVDR`) out into its own section that immediately follows the ideal ratio mask generation section